### PR TITLE
Rename engine.share.level.sub.domain to engine.share.level.subdomain

### PR DESCRIPTION
**Pull request by yaooqinn**
### _Why are the changes needed?_

`subdomain` is a widely used term in computer science

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @yanghua @pan3793

File: kyuubi-server/src/test/scala/org/apache/kyuubi/engine/EngineRefSuite.scala [hunk 1]
```diff
@@ -66,28 +66,32 @@ class EngineRefSuite extends KyuubiFunSuite {
...
     val appName = EngineRef(conf, user, id)
     assert(appName.engineSpace === ZKPaths.makePath(s"kyuubi_$USER", user))
     assert(appName.defaultEngineName === s"kyuubi_${USER}_${user}_${id.identifier}")
 
-    conf.set(KyuubiConf.ENGINE_SHARE_LEVEL_SUB_DOMAIN.key, "abc")
-    val appName2 = EngineRef(conf, user, id)
-    assert(appName2.engineSpace ===
-      ZKPaths.makePath(s"kyuubi_$USER", user, "abc"))
-    assert(appName2.defaultEngineName === s"kyuubi_${USER}_${user}_abc_${id.identifier}")
+    Seq(KyuubiConf.ENGINE_SHARE_LEVEL_SUBDOMAIN,
```
Comment (yaooqinn) [inline kyuubi-server/src/test/scala/org/apache/kyuubi/engine/EngineRefSuite.scala:74]: keep a test with the legacy conf

**Comment by yanghua** [comment]
test failed: `Some("abc") did not equal None (EngineRefSuite.scala:112)`

**Review by yanghua — APPROVED** [review]
LGTM

File: kyuubi-server/src/main/scala/org/apache/kyuubi/engine/EngineRef.scala [hunk 2]
```diff
@@ -112,7 +112,7 @@ private[kyuubi] class EngineRef private(conf: KyuubiConf, user: String, sessionI
   @VisibleForTesting
```
Comment (cfmcgrady) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/engine/EngineRef.scala:112]: nit: also update the comment?
```
   *   /$serverSpace_USER/$user[/$subdomain]
```

Comment (yaooqinn) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/engine/EngineRef.scala:112]: nice catch

**Pull request closed without merging**